src/SpectatorView.Native/SpectatorView.Compositor/UnityPlugin/UnityCompositorInterface.cpp

```diff
@@ -17,33 +17,47 @@
 
 #define UNITYDLL EXTERN_C __declspec(dllexport)
 
-static CompositorInterface* ci = NULL;
+static CompositorInterface* ci = nullptr;
 static bool isRecording = false;
 static bool videoInitialized = false;
 
-static BYTE* colorBytes = new BYTE[FRAME_BUFSIZE];
-static BYTE* holoBytes = new BYTE[FRAME_BUFSIZE];
+static BYTE* colorBytes = new BYTE[FRAME_BUFSIZE_RGBA];
+static BYTE* holoBytes = new BYTE[FRAME_BUFSIZE_RGBA];
 
 #define NUM_VIDEO_BUFFERS 10
 
 static byte** videoBytes = nullptr;
 static int videoBufferIndex = 0;
 
-void AllocateVideoBuffers()
+void AllocateVideoBuffers(VideoRecordingFrameLayout frameLayout)
 {
     if (videoBytes != nullptr)
         return;
 
     videoBytes = new byte*[NUM_VIDEO_BUFFERS];
 
-    for (int i = 0; i < NUM_VIDEO_BUFFERS; i++)
+    int frameBufferSize;
+    if (frameLayout == VideoRecordingFrameLayout::Quad)
     {
 #if HARDWARE_ENCODE_VIDEO
```

Comment:
>HARDWARE_ENCODE_VIDEO [](start = 4, length = 21)

What influences this macro? Is this something that could change dynamically based on switching capture cards?

Reply:
Currently HARDWARE_ENCODE_VIDEO is an option that is hardcoded to TRUE in CompositorShared.h. A future task might be to allow changing back and forth. I think hardware encoding is pretty much required to be able to encode 4K video in realtime though.